include/swift-c/DependencyScan/DependencyScan.h

```diff
@@ -25,204 +25,230 @@
-ds_get_swift_textual_detail_command_line(ds_module_details_t details);
+SWIFTSCAN_PUBLIC swiftscan_string_set_t *
+swiftscan_swift_textual_detail_get_command_line(
+    swiftscan_module_details_t details);
 
-DEPSCAN_PUBLIC ds_string_set_t *
-ds_get_swift_textual_detail_extra_pcm_args(ds_module_details_t details);
+SWIFTSCAN_PUBLIC swiftscan_string_set_t *
+swiftscan_swift_textual_detail_get_extra_pcm_args(
+    swiftscan_module_details_t details);
 
-DEPSCAN_PUBLIC ds_string_t
-ds_get_swift_textual_detail_context_hash(ds_module_details_t details);
+SWIFTSCAN_PUBLIC swiftscan_string_t
+swiftscan_swift_textual_detail_get_context_hash(
+    swiftscan_module_details_t details);
 
-DEPSCAN_PUBLIC bool
-ds_get_swift_textual_detail_is_framework(ds_module_details_t details);
+SWIFTSCAN_PUBLIC bool swiftscan_swift_textual_detail_get_is_framework(
+    swiftscan_module_details_t details);
 
 //=== Swift Binary Module Details query APIs ------------------------------===//
 
-DEPSCAN_PUBLIC ds_string_t
-ds_get_swift_binary_detail_compiled_module_path(ds_module_details_t details);
+SWIFTSCAN_PUBLIC swiftscan_string_t
+swiftscan_swift_binary_detail_get_compiled_module_path(
+    swiftscan_module_details_t details);
 
-DEPSCAN_PUBLIC ds_string_t
-ds_get_swift_binary_detail_module_doc_path(ds_module_details_t details);
+SWIFTSCAN_PUBLIC swiftscan_string_t
+swiftscan_swift_binary_detail_get_module_doc_path(
+    swiftscan_module_details_t details);
 
-DEPSCAN_PUBLIC ds_string_t ds_get_swift_binary_detail_module_source_info_path(
-    ds_module_details_t *details);
+SWIFTSCAN_PUBLIC swiftscan_string_t
+swiftscan_swift_binary_detail_get_module_source_info_path(
+    swiftscan_module_details_t *details);
 
 //=== Swift Placeholder Module Details query APIs -------------------------===//
 
-DEPSCAN_PUBLIC ds_string_t ds_get_swift_placeholder_detail_compiled_module_path(
-    ds_module_details_t details);
+SWIFTSCAN_PUBLIC swiftscan_string_t
+swiftscan_swift_placeholder_detail_get_compiled_module_path(
+    swiftscan_module_details_t details);
 
-DEPSCAN_PUBLIC ds_string_t
-ds_get_swift_placeholder_detail_module_doc_path(ds_module_details_t details);
+SWIFTSCAN_PUBLIC swiftscan_string_t
+swiftscan_swift_placeholder_detail_get_module_doc_path(
+    swiftscan_module_details_t details);
 
-DEPSCAN_PUBLIC ds_string_t
-ds_get_swift_placeholder_detail_module_source_info_path(
-    ds_module_details_t details);
+SWIFTSCAN_PUBLIC swiftscan_string_t
+swiftscan_swift_placeholder_detail_get_module_source_info_path(
+    swiftscan_module_details_t details);
 
 //=== Clang Module Details query APIs -------------------------------------===//
 
-DEPSCAN_PUBLIC ds_string_t
-ds_get_clang_detail_module_map_path(ds_module_details_t details);
+SWIFTSCAN_PUBLIC swiftscan_string_t
+swiftscan_clang_detail_get_module_map_path(swiftscan_module_details_t details);
+
+SWIFTSCAN_PUBLIC swiftscan_string_t
+swiftscan_clang_detail_get_context_hash(swiftscan_module_details_t details);
```

Comment:
These accessors that return `swiftscan_string_t` or `swiftscan_string_set_t *` are not clear about ownership.  I see you have `swiftscan_string_dispose` and `swiftscan_string_set_dispose` methods, but the implementation looks like they are returned _unowned_.  If not every API is consistent about whether the caller needs to dispose of the result, I recommend putting it in the doc comment for the accessors.

Reply:
Resolved by tying the lifetime of the string to the object it is associated with. `swiftscan_string_dispose` and `swiftscan_string_set_dispose` are no longer part of the public API.

Reply:
In general, [I propose we adopt Foundation-style ownership signals](https://developer.apple.com/library/archive/documentation/CoreFoundation/Conceptual/CFMemoryMgmt/Concepts/Ownership.html#//apple_ref/doc/uid/20001148-CJBEJBHH) here

Reply:
> In general, [I propose we adopt Foundation-style ownership signals](https://developer.apple.com/library/archive/documentation/CoreFoundation/Conceptual/CFMemoryMgmt/Concepts/Ownership.html#//apple_ref/doc/uid/20001148-CJBEJBHH) here

@CodaFi, thank you taking a look at this PR and proposing this convention. 
The API now consistently follows this naming scheme, providing a `create` and `dispose` functions for all objects that the library's clients may take ownership of.